src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resizedIfNeeded();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(index + " are incorect");
+        }
+        resizedIfNeeded();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        resizedIfNeeded();
+        int index = size;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i), index++);
+        }
```

Comment:
may be simplified
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```
our method add(T value) will resize when required and will add properly in the end